activejob/lib/active_job/test_helper.rb

```diff
@@ -664,7 +664,7 @@ def flush_enqueued_jobs(only: nil, except: nil, queue: nil, at: nil)
         enqueued_jobs_with(only: only, except: except, queue: queue, at: at) do |payload|
           queue_adapter.enqueued_jobs.delete(payload)
           queue_adapter.performed_jobs << payload
-          instantiate_job(payload).perform_now
+          instantiate_job(payload, skip_deserialize_arguments: true).perform_now
```

Comment:
This seems like the only place that calls `instantiate_job`.
Could we maybe even remove the call to `deserialize_arguments_if_needed`?

Reply:
@p8 `instantiate_job` is also called in:
- [`assert_enqueued_with`](https://github.com/rails/rails/blob/main/activejob/lib/active_job/test_helper.rb#L423)
- [`assert_performed_with`](https://github.com/rails/rails/blob/main/activejob/lib/active_job/test_helper.rb#L514)

I was afraid of removing the call of `deserialize_arguments_if_needed` completely because we might have some tests triggered from the 2 methods above which relies on that because the 2 methods above doesn't call `perform_now` thereafter.

You think we can safely remove it anyways?
Thanks!

Reply:
@p8 pinging you incase you lost the notification 🙇